internal/pool/conn.go

```diff
@@ -64,14 +64,17 @@ func (cn *Conn) RemoteAddr() net.Addr {
 }
 
 func (cn *Conn) WithReader(
-	ctx context.Context, timeout time.Duration, fn func(rd *proto.Reader) error,
+	ctx context.Context, timeout time.Duration, fn []func(rd *proto.Reader) error,
```

Comment:
I think this would make our library, immediately much slower. Adding a loop here - means parsing speeds increase.